client/cmd/dexc-desktop/syncserver.go

```
@@ -0,0 +1,163 @@
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/app"
+	"decred.org/dcrdex/dex"
+)
+
+const (
+	syncFilename = "syncfile"
+)
+
+// getSyncAddress: Get the address stored in the syncfile. If the file doesn't
+// exist, an empty string is returned with no error.
+func getSyncAddress(syncDir string) (addr string, err error) {
+	syncFile := filepath.Join(syncDir, syncFilename)
+	addrB, err := os.ReadFile(syncFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return "", err
+		}
+		return "", nil
+	}
+	return string(addrB), nil
+}
+
+// sendKillSignal yeets a kill command to any address in the sync file.
+func sendKillSignal(syncDir string) {
+	addr, err := getSyncAddress(syncDir)
+	if err != nil {
+		log.Errorf("Error getting sync address for kill signal: %v", err)
+		return
+	}
+	if addr == "" {
+		log.Errorf("No sync address found for kill signal: %v", err)
+		return
+	}
+	resp, err := http.Get("http://" + addr + "/kill")
+	if err != nil {
+		log.Errorf("Error sending kill signal to sync server: %v", err)
+		return
+	}
+	if resp == nil {
+		log.Errorf("nil kill response")
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("Unexpected response code from kill signal send: %d (%s)", resp.StatusCode, http.StatusText(resp.StatusCode))
+	}
+}
+
+// synchronize attempts to determine the state of any other dexc-desktop
+// instances that are running. If none are found, startServer will be true.
+// If another instance is found, close will be true, and this instance of
+// dexc-desktop should exit immediately.
+func synchronize(syncDir string) (startServer, close bool, err error) {
+	addr, err := getSyncAddress(syncDir)
+	if err != nil {
+		return false, false, err
+	}
+	if addr == "" {
+		// Just start the server then.
+		return true, false, nil
+	}
+	resp, err := http.Get("http://" + addr)
+	if err == nil && resp.StatusCode == http.StatusOK {
+		// Other instance will open the window.
+		return false, true, nil
```

If `err == nil` but the response code is not 200, we should return an error, not signal to start the server since one is already running on `addr`.